src/libs/ReportUtils.js

```diff
@@ -512,16 +512,38 @@ function navigateToDetailsPage(report) {
 }
 
 /**
- * Generate a random reportID between 98000000 (the number of reports before the switch from sequential to random)
- * and the maximum safe integer of js (53 bits aka 9,007,199,254,740,991)
+ * Generate a random reportID up to 53 bits aka 9,007,199,254,740,991 (Number.MAX_SAFE_INTEGER).
+ * There were approcimately 98,000,000 reports with sequential IDs generated before we started using this approach, those make up roughly one billionth of the space for these numbers,
+ * so we live with the 1 in a billion chance of a collision with an older ID until we can switch to 64-bit IDs.
  *
  * In a test of 500M reports (28 years of reports at our current max rate) we got 20-40 collisions meaning that
  * this is more than random enough for our needs.
  *
  * @returns {Number}
  */
 function generateReportID() {
-    return Math.floor(Math.random() * (Number.MAX_SAFE_INTEGER - 98000000)) + 98000000;
+    // Generate a random 32-bit number. This works fine, and starts us building a random 53-bit number.
+    let result = Math.floor(Math.random() * (2 ** 32));
+
+    // Now generate another random number. We'll use this for the remaining 21 bits of randomness.
+    let extraBits = Math.floor(Math.random() * (2 ** 32));
+
+    // Add each of the remaining 21 bits to the end of `result`.
+    for (let i = 0; i < 21; i++) {
+        // Shift left by one, but don't use bit shifts: they truncate to 32-bits.
+        result *= 2;
+
+        // If extraBits is odd, meaning the lowest bit is set, do the same to the result, without bitwise operators.
+        if (extraBits % 2 === 0) {
```

Comment:
Isn't this checking if extraBits is even?

Reply:
Oops, yes, fixed. This actually works identically either way since the source of the data is random. If we add the exact opposite of a random number, it's an equally random number, but I fixed it to be clear.